File: src/main/scala/project1/HousePriceRegression.scala
package project1

//import scalation.mathstat._
import scalation.theory._

object HousePriceRegression {
  // previously used raw dataset, but now use clean after running feature selection algorithms on the raw dataset
  val filePath = "project1/house_price_regression_dataset_cleaned.csv"

  val dataName = "HousePriceRegression"
  val data: Dataset = Dataset(dataName, filePath, 7, Array(0, 1, 2, 3, 4, 5), 6)

  @main def run(): Unit = {
    val utils = new Project1Utils(data.x, data.y, data.fname)
    //utils.runLinearRegressions()
    //utils.runRidgeRegressions()
    utils.runLassoRegressions()
//    utils.runTransformedRegressions()
//    utils.runSymRidgeRegression()
//    utils.runForwardSelect()
//    utils.runBackwardsElimination()
//    utils.runStepwiseSelect()
  }
}

File: notebooks/project1/house_price_regression/house_price_regression.py
import pandas as pd
import os

from project1.project1_utils import Project1Utils


def main():
    file_path = os.path.expanduser('../../../data/project1/house_price_regression_dataset_cleaned.csv')

    df = pd.read_csv(file_path)

    target_col = df.columns[-1]
    feature_cols = df.columns[:-1]

    X = df[feature_cols]
    y = df[target_col]

    # Run
    utils = Project1Utils(X, y)
    utils.run_linear_regressions()
    utils.run_ridge_regressions(alpha=0.5)
    utils.run_lasso_regressions(alpha=0.5)
    utils.run_transformed_regressions()
    utils.run_sym_ridge_regression()

if __name__ == "__main__":
    main()